astropy/utils/data.py

```diff
@@ -871,10 +871,72 @@ def compute_hash(localfn):
     return h.hexdigest()
 
 
+def get_pkg_data_path(*path, package=None):
+    """Get path from source-included data directories.
+
+    Parameters
+    ----------
+    *path : str
+        Name/location of the desired data file/directory.
+        May be a tuple of strings for ``os.path`` joining.
+
+    package : str or `None`, optional, keyword only
+        If specified, look for a file relative to the given package, rather
+        than the calling module's package.
+
+    Returns
+    -------
+    path : str
+        Name/location of the desired data file/directory.
+
+    Raises
+    ------
+    ImportError
+        Given package or module is not importable.
+    """
+    if package is None:
+        module = find_current_module(1, finddiff=['astropy.utils.data', 'contextlib'])
+        if module is None:
+            # not called from inside an astropy package.  So just pass name
+            # through
+            return os.path.join(*path)
+
+        if not hasattr(module, '__package__') or not module.__package__:
+            # The __package__ attribute may be missing or set to None; see
+            # PEP-366, also astropy issue #1256
+            if '.' in module.__name__:
+                package = module.__name__.rpartition('.')[0]
+            else:
+                package = module.__name__
+        else:
+            package = module.__package__
+    else:
+        module = resolve_name(package)
+
+    rootpkgname = package.partition('.')[0]
+
+    rootpkg = resolve_name(rootpkgname)
+
+    module_path = os.path.dirname(module.__file__)
+    path = os.path.join(module_path, *path)
+
+    root_dir = os.path.dirname(rootpkg.__file__)
+    if not _is_inside(path, root_dir):
```

Comment:
This line is a bug since `path` is now a tuple.

Reply:
it shouldn't be a tuple since the ``path = os.path.join(module_path, *path)`` make it a string. I'm changing it to ``full_path``